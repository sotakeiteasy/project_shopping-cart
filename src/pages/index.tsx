import Home from "./home/home";
import Catalog from "./catalog/catalog";
import Cart from "./cart/cart";

export { Home, Catalog, Cart };
