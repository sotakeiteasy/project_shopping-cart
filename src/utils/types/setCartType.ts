import { Product } from "./product";
export type SetCartType = React.Dispatch<React.SetStateAction<Product[]>>;
