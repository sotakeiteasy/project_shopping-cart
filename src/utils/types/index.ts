import { Product } from "./product";
import { SetCartType } from "./setCartType";

export type { Product, SetCartType };
