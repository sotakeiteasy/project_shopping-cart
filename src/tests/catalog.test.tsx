import { render, screen } from "@testing-library/react";
import { Catalog } from "@/pages";
import { Product } from "@/utils/types";

describe("render product", () => {
    vi.mock("@/components", () => ({
        ProductCard: (product: Product) => <div>{product.title}</div>,
    }));

    it("loader if no products", async () => {
        global.fetch = vi.fn().mockResolvedValue({
            json: async () => [],
        });

        render(<Catalog />);

        expect(await screen.findByText(/loading data/i)).toBeInTheDocument();
    });

    it("no loader if products", async () => {
        global.fetch = vi.fn().mockResolvedValue({
            json: async () => [{ id: 1, title: 'Test' }],
        });

        render(<Catalog />);

        expect(await screen.findByText(/loading data/i)).not.toBeInTheDocument();
    });
});

describe("render catalog correctly", () => {
    it("renders catalog", async () => {
        const { container } = render(<Catalog />);
        expect(container).toMatchSnapshot()
    })

    it("renders main info", async () => {
        render(<Catalog />);

        expect(screen.getByRole("button", { name: /all/i })).toBeInTheDocument()
        expect(screen.getByRole("button", { name: /new/i })).toBeInTheDocument()
        expect(screen.getByRole("button", { name: /sale/i })).toBeInTheDocument()
        expect(screen.getByRole('heading', { name: /catalog/i }).textContent).toMatch('Catalog')
    })
})