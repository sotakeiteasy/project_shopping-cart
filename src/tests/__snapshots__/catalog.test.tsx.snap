// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`render catalog correctly > renders catalog 1`] = `
<div>
  <main
    class="_catalog_4f6bc5"
  >
    <h2>
      Catalog
    </h2>
    <section
      class="_filters_4f6bc5"
    >
      <button>
        All
      </button>
      <button>
        New
      </button>
      <button>
        Sale
      </button>
    </section>
    <section
      class="_products_4f6bc5"
    >
      <div>
        Loading data...
      </div>
    </section>
    <section
      class="_infoSection_4f6bc5"
    >
      <h3>
        Shopping Info
      </h3>
      <ul>
        <li>
          Free delivery over $50
        </li>
        <li>
          14-day returns & exchanges
        </li>
        <li>
          Support: support@example.com
        </li>
      </ul>
    </section>
  </main>
</div>
`;
